planning/behavior_path_planner/src/behavior_path_planner_node.cpp

```diff
@@ -445,41 +465,56 @@ BehaviorTreeManagerParam BehaviorPathPlannerNode::getBehaviorTreeManagerParam()
 void BehaviorPathPlannerNode::waitForData()
 {
   // wait until mandatory data is ready
+  mutex_.lock();  // for planner_data_
   while (!planner_data_->route_handler->isHandlerReady() && rclcpp::ok()) {
+    mutex_.unlock();
     RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), 5000, "waiting for route to be ready");
     rclcpp::spin_some(this->get_node_base_interface());
     rclcpp::Rate(100).sleep();
+    mutex_.lock();
   }
 
   while (rclcpp::ok()) {
     if (planner_data_->dynamic_object && planner_data_->self_odometry) {
       break;
     }
+    mutex_.unlock();
     RCLCPP_WARN_THROTTLE(
       get_logger(), *get_clock(), 5000,
       "waiting for vehicle pose, vehicle_velocity, and obstacles");
     rclcpp::spin_some(this->get_node_base_interface());
     rclcpp::Rate(100).sleep();
+    mutex_.lock();
   }
 
   self_pose_listener_.waitForFirstPose();
   planner_data_->self_pose = self_pose_listener_.getCurrentPose();
+  mutex_.unlock();
 }
 
 void BehaviorPathPlannerNode::run()
 {
   RCLCPP_DEBUG(get_logger(), "----- BehaviorPathPlannerNode start -----");
+  mutex_.lock();  // for planner_data_
 
   // update planner data
   updateCurrentPose();
 
+  // NOTE: planner_data must not be referenced for multithreading
+  const auto planner_data = planner_data_;
+  mutex_.unlock();
+
   // run behavior planner
-  const auto output = bt_manager_->run(planner_data_);
+  const auto output = bt_manager_->run(planner_data);
 
   // path handling
   const auto path = getPath(output);
   const auto path_candidate = getPathCandidate(output);
+
+  // update planner data
+  mutex_.lock();  // for planner_data_
   planner_data_->prev_output_path = path;
+  mutex_.unlock();
```

Comment:
Here, not planner_data_, but planner_data.

Reply:
without mutex

Reply:
The lock of mutex is required here for value-assignment to member variables to planner_data_